error_handling/either_type.md

```diff
@@ -0,0 +1,36 @@
+#EitherType
+By using dartz package, we always return Either from a repository method where left part is our own Failure class and right part is expected result if execution of the method ended with success. By using Either, caller of the repository methods doesn't have to wrap each call in try catch block and is just able to wait for the result of the method and fold it to be prepared for both outcomes, fail and success.\
```

Comment:
isto je motivacija sto na ovaj nacin caller repository metode mora hendlati errore, ne moze zaboraviti to jer je u samnom povratnom tipu metode definirano da je moguce da ce se desiti greska... isto cini kod samodokumentirajucim